packages/flutter/test/material/expansion_tile_test.dart

```diff
@@ -109,15 +109,15 @@ void main() {
     expect(getHeight(topKey), getHeight(collapsedKey) - 2.0);
     expect(getHeight(topKey), getHeight(defaultKey) - 2.0);
 
-    BoxDecoration expandedContainerDecoration = getContainer(expandedKey).decoration! as BoxDecoration;
+    ShapeDecoration expandedContainerDecoration = getContainer(expandedKey).decoration! as ShapeDecoration;
     expect(expandedContainerDecoration.color, Colors.red);
-    expect(expandedContainerDecoration.border!.top.color, dividerColor);
-    expect(expandedContainerDecoration.border!.bottom.color, dividerColor);
+    expect((expandedContainerDecoration.shape as Border?)!.top.color, dividerColor);
```

Comment:
Why are all of these casts necessary? Is there a way to avoid them?

Reply:
As `expandedContainerDecoration.shape` is a `ShapeBorder` we need to cast it to `Border` to get the top, and bottom colors for the test.

Mainly cause `expandedContainerDecoration` is `ShapeDecoration`